### PR TITLE
Select: Support placeholder prop on Trigger

### PR DESCRIPTION
**Pull request by mirka**
## What?

Adds a `placeholder` prop to the `Select.Trigger` primitive, defaulting to `Select`, and aligns placeholder styling with Base UI's built-in placeholder state.

Also updates the Select stories to demonstrate a custom placeholder and a clearable `null` value option.

## Why?

Base UI now provides official placeholder handling for Select. Using that state directly keeps the wrapper behavior aligned with upstream and avoids relying on an empty-string item value as a placeholder convention.

## How?

- Passes `placeholder` from `Select.Trigger` to Base UI's `Select.Value`.
- Uses Base UI's `data-placeholder` attribute for placeholder styling.
- Allows `Select.Item` values to be `null`, matching Base UI's clearable item pattern.
- Adds focused unit coverage for default placeholder, custom placeholder, and selected-value states.

## Testing Instructions

3. Open Storybook and review `Design System/Components/Form/Primitives/Select`.
4. Confirm the custom placeholder story shows custom text before selection.
5. Confirm the null value option story can clear the selected value from the popup.

## Screenshots or screencast

Select with placeholder

### CODE REVIEW

File: packages/ui/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 ### Enhancements
 
+-   `Select`: Add a `placeholder` prop to `Select.Trigger`, and support `null` item values for clearable placeholder options. A default placeholder is now applied when no initial value is specified ([#78076](https://github.com/WordPress/gutenberg/pull/78076)).
```
Comment (ciampo) [inline packages/ui/CHANGELOG.md]: Not sure if we want to mention a couple of behavioural changes:

- Consumers relying on "item with an empty string value used as a placeholder" will lose the placeholder styling
- Previously, with no items and no selected value, the trigger rendered empty. it now renders "Select"

Slightly breaking, but also enhancements.

File: packages/ui/src/form/primitives/select/test/index.test.tsx
Comment (ciampo) [inline packages/ui/src/form/primitives/select/test/index.test.tsx:1]: Should we add a test for the null-item clearable flow described in the docstring (open the popup, select a value, click the null item, assert the trigger reverts to the placeholder with `data-placeholder`)?

File: packages/ui/src/form/primitives/select/test/index.test.tsx [hunk 2]
```diff
@@ -4,6 +4,56 @@ import { createRef } from '@wordpress/element';
...
+				<Select.Popup>
+					<Select.Item value="Item 1" />
+				</Select.Popup>
+			</Select.Root>
+		);
+
+		const value = screen.getByText( 'Item 1' );
+
+		expect( screen.getByRole( 'combobox' ) ).toHaveTextContent( 'Item 1' );
+		expect( value ).not.toHaveAttribute( 'data-placeholder' );
```
Comment (ciampo) [inline packages/ui/src/form/primitives/select/test/index.test.tsx]: Potentially stricter / more correct?

```ts
const trigger = screen.getByRole( 'combobox' );
expect( trigger ).toHaveTextContent( 'Item 1' );
expect( within( trigger ).getByText( 'Item 1' ) ).not.toHaveAttribute( 'data-placeholder' );
```

File: packages/ui/src/utils/css/select-trigger.module.css [hunk 3]
```diff
@@ -48,7 +48,7 @@
 		text-overflow: ellipsis;
 		white-space: nowrap;
 
-		&.is-placeholder {
+		&[data-placeholder] {
```
Comment (ciampo) [inline packages/ui/src/utils/css/select-trigger.module.css:51]: I feel like we should flag more clearly in `Select.Trigger` that the trigger contents will get "placeholder styles" when there's no value selected, even if they supply a custom child function?

**Review by ciampo — APPROVED** [review]
LGTM 🚀

File: packages/ui/src/form/primitives/select/test/index.test.tsx
Comment (mirka) [reply packages/ui/src/form/primitives/select/test/index.test.tsx:1]: This is just standard upstream behavior, so I opted to omit testing on our end.
